web/html/src/manager/virtualization/guests/console/guests-console.js

```diff
@@ -53,6 +62,10 @@ class GuestsConsole extends React.Component<Props, State> {
     this.pageUnloading = false;
     this.openVirtSocket();
 
+    // Schedule token refresh a minute before its expiration
+    const tokenExp = getTokenExpiration(props.token);
+    this.intervalId = window.setInterval(this.refreshToken, tokenExp - Date.now() - 60000);
```

Comment:
Is this ok? IIUC, this is to guess the life span of the token, but there can be a difference between issuing the token and `Date.now()`. Can this difference grow big under some circumstances? What happens if the client clock is off by one hour? Wouldn't it be better to build the life span attribute to the token directly (if it's not there yet)?

Reply:
The hacky way could be to hard code a refresh every 4 minutes (5 minutes token refresh - 1 minute). We don't have the token lifespan in the token yet

Reply:
Awesome!